lib/mapUtils.js

```diff
@@ -43,3 +43,156 @@ export async function getProductData(store) {
+      // e.g. timeAndDaySplit = ['9am-8pm', 'Tu-Sat']
+      const timeAndDaySplit = timeAndDayPair.split(' ');
+      // e.g. 9am-8pm
+      const time = timeAndDaySplit[0];
+      // e.g. [Tu, Sat] or [Mon]
+      const days = timeAndDaySplit[1].split('-');
+
+      // Case: Time corresponds to one day
+      if (days.length === 1) {
+        hoursPerDayDict[daysOfTheWeek[days[0]]] = time;
+      } else if (days.length === 2) {
+        const firstDay = days[0];
+        const lastDay = days[1];
+
+        // ['Sun', 'Mon', 'Tues', ...]
+        const indexFirstDay = daysOfTheWeekAbbrev.indexOf(firstDay);
+        const indexLastDay = daysOfTheWeekAbbrev.indexOf(lastDay);
+
+        let numDaysInInterval;
+
+        if (indexLastDay > indexFirstDay) {
+          // Case: Interval is Sun-Thurs, 0-4
+          numDaysInInterval = indexLastDay - indexFirstDay + 1;
+        } else {
+          // Case: Interval is Thurs-Sun, 4-0
+          numDaysInInterval = indexLastDay - indexFirstDay + 8;
+        }
+
+        [...Array(numDaysInInterval).keys()].map((i, _) => {
+          const dayIndex = (indexFirstDay + i) % 7;
+          hoursPerDayDict[daysOfTheWeekFull[dayIndex]] = time;
+        });
+      } else {
+        // Shouldn't ever have a case with more than two days in an interval.
+        console.error(
+          '[mapUtils] constructHoursDict: Incorrect store hours formatting in Airtable.'
+        );
+      }
+    });
+  }
+  return hoursPerDayDict;
+}
+
+export function computeStoreOpen(hours) {
+  if (hours === 'Open 24/7' || hours === 'Store hours unavailable') {
+    return hours;
+  }
+
+  const daysOfTheWeek = {
+    Sun: 'Sunday',
+    Mon: 'Monday',
+    Tu: 'Tuesday',
+    Wed: 'Wednesday',
+    Thurs: 'Thursday',
+    Fri: 'Friday',
+    Sat: 'Saturday',
+  };
+  const daysOfTheWeekFull = Object.values(daysOfTheWeek);
+  const today = new Date();
+  // currentTime is minutes since 12am
+  const currentTime = today.getHours() * 60 + today.getMinutes();
+
+  const todaysDayIndex = today.getDay();
+  const todaysDay = daysOfTheWeekFull[todaysDayIndex];
+
+  const hoursPerDayDict = constructHoursDict(hours);
+
+  // e.g. "8am-8pm"
+  const todaysStoreHours = hoursPerDayDict[todaysDay];
+  const hoursSplit = todaysStoreHours.split('-');
+  const openingTime = hoursSplit[0]; // e.g. "8:30am"
+  const closingTime = hoursSplit[1]; // e.g. "8:30pm"
+
+  const openingSuffix = openingTime.slice(-2);
+  const openingHourMin = openingTime.slice(0, -2);
+  const openingHourMinSplit = openingHourMin.split(':');
+  let openingHour = openingHourMinSplit[0];
+  const openingMin = openingHourMinSplit[1];
+
+  const closingSuffix = closingTime.slice(-2);
+  const closingHourMin = closingTime.slice(0, -2);
+  const closingHourMinSplit = closingHourMin.split(':'); // e.g. ["8", "30"]
+  let closingHour = closingHourMinSplit[0];
+  const closingMin = closingHourMinSplit[1];
+
+  // Convert to 24 hour time
+  if (openingSuffix === 'pm') {
```

Comment:
In the airtable it looked like the hours were inputted in the form of:
`7:30am-10pm Daily`
`7am-10pm Mon-Sat, 9am-9pm Sun`

Would it be safe for me to assume that it would follow the `timeInterval Day` format (regardless of weird spacing issues)? I haven't accounted for the ones you linked in the picture